snuba/request/request_settings.py

```diff
@@ -30,6 +30,10 @@ def get_consistent(self) -> bool:
     def get_debug(self) -> bool:
         pass
 
+    @abstractmethod
+    def get_parent_api(self) -> str:
+        pass
+
```

Comment:
I don't think query settings is the right place for this. 
RequestSettings are supposed to allow the query pipeline to change its logic depending how the user wants the query to be run.
They allow the user to pick the level of consistency, they contain the rate limiter config, etc. 
The parent API should never be used to change the logic of a query processor or any step in the query execution. That would mean breaking the separation of responsibility between sentry and snuba acts differently depending where the query is executed in sentry, thus coupling the snuba query processing to the context in which the query is executed.

I think what we need is some sort of shared observability context. Separate from the request settings, that is possibly opaque to query processors. 

What about reusing a metrics object across the query execution pipeline to record metrics? This would be instead instead of reinstantiating it in every module? This could also guarantee that all relevant context tags (referrer, parent api, consistent, dataset, etc.) are present in all metrics. 
Still individual query processors would not be able to extract this information. 

We could pass it through to query processors, validators, etc instead of creating it from scratch.

Reply:
Yes, I agree with all of what you're saying, and I posted in Slack about adding a "request" context that could contain all of this. It's just a question of timelines: how long are you willing to wait for this data?

Reply:
Well, I guess we do not many options now.